### PR TITLE
[MB-3729] Create update addresses endpoint

### PR DESCRIPTION
**Pull request by shimonacarvalho**
## Description

Abigail added an endpoint for updating addresses, I took the ticket on and split out the service object and added tests.

Endpoint is a prime api endpoint located here:
`/prime/v1/mto-shipments/{mtoShipmentID}/addresses/{addressID}`

Please try the endpoint and tests.

## References

* [MB-3729 Prime API: Create endpoint to update addresses - Jira](https://dp3.atlassian.net/browse/MB-3729)
* [Acceptance testing](https://github.com/transcom/mymove/wiki/Acceptance-Testing-Prime-API-Endpoints)

### CODE REVIEW

File: pkg/handlers/primeapi/internal/payloads/payload_to_model.go [hunk 1]
```diff
@@ -90,12 +91,14 @@ func MTOShipmentModelFromCreate(mtoShipment *primemessages.CreateMTOShipment) *m
...
-		MoveTaskOrderID:     uuid.FromStringOrNil(mtoShipment.MoveTaskOrderID.String()),
-		ShipmentType:        models.MTOShipmentType(mtoShipment.ShipmentType),
-		RequestedPickupDate: &requestedPickupDate,
-		CustomerRemarks:     mtoShipment.CustomerRemarks,
+		MoveTaskOrderID: uuid.FromStringOrNil(mtoShipment.MoveTaskOrderID.String()),
+		ShipmentType:    models.MTOShipmentType(mtoShipment.ShipmentType),
+		CustomerRemarks: mtoShipment.CustomerRemarks,
+	}
+
+	if mtoShipment.RequestedPickupDate != nil {
```
Comment (shimonacarvalho) [inline pkg/handlers/primeapi/internal/payloads/payload_to_model.go:100]: In the course of doing this ticket, I found some issues with RequestedPickupDate. It was validated as required in our code but not marked as required in the yaml.

File: pkg/handlers/primeapi/mto_shipment_address_test.go [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+		suite.Equal(*reqAddress.StreetAddress2, *respAddress.StreetAddress2)
+	}
+	if reqAddress.StreetAddress3 != nil && respAddress.StreetAddress3 != nil {
+		suite.Equal(*reqAddress.StreetAddress3, *respAddress.StreetAddress3)
+	}
+	suite.Equal(*reqAddress.PostalCode, *respAddress.PostalCode)
+	suite.Equal(*reqAddress.State, *respAddress.State)
+	suite.Equal(*reqAddress.City, *respAddress.City)
+
+}
```
Comment (sandy-wright) [inline pkg/handlers/primeapi/mto_shipment_address_test.go:36]: nice, this is super thorough!

File: swagger/prime.yaml [hunk 3]
```diff
@@ -606,7 +680,7 @@ definitions:
       name:
         type: string
         example: Fort Bragg North Station
-      addressId:
+      addressID:
```
Comment (sandy-wright) [inline swagger/prime.yaml:683]: bless 🙏

**Review by sandy-wright — APPROVED** [review]
LGTM - tested a bunch of cases, and it holds up! 🚀

**Pull request merged**